lib/spack/spack/parser.py

```diff
@@ -206,11 +206,15 @@ def tokenize(text: str) -> Iterator[Token]:
     scanner = ALL_TOKENS.scanner(text)  # type: ignore[attr-defined]
     match: Optional[Match] = None
     for match in iter(scanner.match, None):
+        # The following two assertions are to help mypy
+        msg = (
+            "unexpected value encountered during parsing. Please submit a bug report "
```

Comment:
I didn't on purpose, this will be prefixed by:
```
==> Error: unexpected ....
```
when shown to users (and hopefully it'll never be shown :slightly_smiling_face: )